### PR TITLE
Scroll results into view when paginating

### PR DESCRIPTION
**Pull request by jonkafton**
### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4741

### Description (What does it do?)

Scrolls the top of the search results to the center of the page on pagination.

### Screenshots (if appropriate):

https://github.com/mitodl/mit-open/assets/939376/816e3a70-5be4-46d4-8566-dda64becd94d

### How can this be tested?

Navigate to the search page or a unit page and paginate through search results.
- http://od.odl.local:8063/search/
- http://od.odl.local:8063/c/unit/ocw/

### CODE REVIEW

File: frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx [hunk 1]
```diff
@@ -503,6 +503,7 @@ const SearchDisplay: React.FC<SearchDisplayProps> = ({
   setSearchParams,
 }) => {
   const [searchParams] = useSearchParams()
+  const scrollHook = useRef<HTMLInputElement>(null)
```
Comment (rhysyngsun) [inline frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx:506]: I think this should just be `HTMLElement` because it's a `div`

**Review by rhysyngsun — CHANGES_REQUESTED** [review]
Functionality looks good, just one piece of feedback.

File: frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx (hunk 1 above)
Comment (jonkafton) [reply frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx]: Good spot, thanks.

**Review by rhysyngsun — APPROVED** [review]
LGTM

File: frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx [hunk 2]
```diff
@@ -680,7 +681,15 @@ const SearchDisplay: React.FC<SearchDisplayProps> = ({
...
                   page={page}
-                  onChange={(_, newPage) => setPage(newPage)}
+                  onChange={(_, newPage) => {
+                    setPage(newPage)
+                    setTimeout(() => {
+                      scrollHook.current?.scrollIntoView({
+                        block: "center",
+                        behavior: "smooth",
+                      })
+                    }, 0)
```
Comment (ChristopherChudzicki) [inline frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx:691]: Out of curiosity... why wrap this in setTimeout?

Comment (jonkafton) [reply frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx:691]: Despite some investigating, all I can say for sure is that it doesn't work without it. Likely some nuance around the dom element not being available during the old render cycle.

**Pull request merged**